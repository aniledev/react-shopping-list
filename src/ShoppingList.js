import React from "react";

export default class ShoppingList extends React.Component {
  render() {
    return <div></div>;
  }
}
