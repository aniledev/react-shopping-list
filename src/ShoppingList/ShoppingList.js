import React from "react";

// an individual item within the list that contains a check button and delete button. There will be 0 or more of these
export default class ShoppingList extends React.Component {
  render() {
    return <div></div>;
  }
}
