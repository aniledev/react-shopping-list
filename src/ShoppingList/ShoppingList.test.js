import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ShoppingList from "./ShoppingList";

describe("ShoppingList Component", () => {
  // start with a smoke test inside a describe block; does the component render properly without errors
  it("should render without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ShoppingList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // if there are no items in the array, then nothing should render
  it("should render empty given no items", () => {
    const wrapper = shallow(<ShoppingList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it should render multiple items when supplied with a proper array of items
  it("should render ShoppingItem when supplied with an items prop", () => {
    const testItems = [
      { name: "test item 1", checked: false },
      { name: "another test item", checked: true },
      { name: "test item number 3", checked: false },
    ];
    const wrapper = shallow(<ShoppingList items={testItems} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
