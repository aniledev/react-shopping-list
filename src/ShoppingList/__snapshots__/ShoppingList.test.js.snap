// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList Component should render ShoppingItems when supplied with an items prop 1`] = `
<ul>
  <ShoppingItem
    item={
      Object {
        "checked": false,
        "name": "test item 1",
      }
    }
    key="0"
  />
  <ShoppingItem
    item={
      Object {
        "checked": true,
        "name": "another test item",
      }
    }
    key="1"
  />
  <ShoppingItem
    item={
      Object {
        "checked": false,
        "name": "test item number 3",
      }
    }
    key="2"
  />
</ul>
`;

exports[`ShoppingList Component should render empty given no items 1`] = `<ul />`;
